lib/core/utils/cache.js

```diff
@@ -0,0 +1,33 @@
+(function() {
+	'use strict';
+	let _cache = {};
+
+	const cache = {
+		/**
+		 * Set an item in the cache.
+		 * @param {String} key - Name of the key.
+		 * @param {*} value - Value to store.
+		 */
+		set(key, value) {
+			_cache[key] = value;
+		},
+
+		/**
+		 * Retrieve an item from the cache.
+		 * @param {String} key - Name of the key the value was stored as.
+		 * @returns {*} The item stored
+		 */
+		get(key) {
+			return _cache[key];
+		},
+
+		/**
+		 * Clear the cache.
+		 */
+		clear() {
+			_cache = {};
+		}
+	};
+
+	axe.utils.cache = cache;
```

Comment:
I'm a little nervous about adding this to "public" namespace. It's very hard for us to deprecate public APIs right now, and I'm worried that this could make that more difficult.

For example, what if a custom rule starts using this to store some sort of arbitrary data?